include/engine/guidance/assemble_leg.hpp

```diff
@@ -181,9 +181,10 @@ inline RouteLeg assembleLeg(const datafacade::BaseDataFacade &facade,
                                   summary_array.end(),
                                   facade.GetNameForID(summary_array.front()),
                                   [&facade](std::string previous, const std::uint32_t name_id) {
-                                      if (name_id != 0)
+                                      auto next_name = facade.GetNameForID(name_id);
+                                      if (name_id != 0 && !next_name.empty())
```

Comment:
`name_id != 0` is a superfluous check now. After all you requested the name already. In addition, we should check if the `name_id` is valid (https://github.com/Project-OSRM/osrm-backend/blob/master/include/util/typedefs.hpp#L81).